packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/data/versioning/handlers/DeleteBranchRestEventHandler.java

```diff
@@ -0,0 +1,85 @@
+package org.eclipse.sirius.web.application.project.data.versioning.handlers;
+
+import java.text.MessageFormat;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.eclipse.sirius.components.collaborative.api.ChangeDescription;
+import org.eclipse.sirius.components.collaborative.api.ChangeKind;
+import org.eclipse.sirius.components.collaborative.api.IEditingContextEventHandler;
+import org.eclipse.sirius.components.collaborative.api.Monitoring;
+import org.eclipse.sirius.components.core.api.ErrorPayload;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IInput;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.DeleteBranchRestInput;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.DeleteBranchRestSuccessPayload;
+import org.eclipse.sirius.web.application.project.data.versioning.services.api.IProjectDataVersioningRestService;
+import org.eclipse.sirius.web.domain.services.api.IMessageService;
+import org.springframework.stereotype.Service;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import reactor.core.publisher.Sinks.Many;
+import reactor.core.publisher.Sinks.One;
+
+/**
+ * Handler used to execute the "deleteBranch" REST API.
+ *
+ * @author arichard
+ */
+@Service
+public class DeleteBranchRestEventHandler implements IEditingContextEventHandler {
+
+    private final IProjectDataVersioningRestService projectDataVersioningRestService;
+
+    private final IMessageService messageService;
+
+    private final Counter counter;
+
+    public DeleteBranchRestEventHandler(IProjectDataVersioningRestService projectDataVersioningRestService, IMessageService messageService, MeterRegistry meterRegistry) {
+        this.projectDataVersioningRestService = Objects.requireNonNull(projectDataVersioningRestService);
+        this.messageService = Objects.requireNonNull(messageService);
+
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+    }
+
+    @Override
+    public boolean canHandle(IEditingContext editingContext, IInput input) {
+        return input instanceof DeleteBranchRestInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IInput input) {
+        this.counter.increment();
+
+        String message = this.messageService.invalidInput(input.getClass().getSimpleName(), DeleteBranchRestInput.class.getSimpleName());
+        IPayload payload = new ErrorPayload(input.id(), message);
+        if (input instanceof DeleteBranchRestInput restInput) {
+            var branch = this.projectDataVersioningRestService.deleteBranch(editingContext, restInput.branchId());
```

Comment:
While it's not an issue today, there's one more hint that this kind of API will have to evolve in the future. I've never noticed it but you are on one hand giving access to the editing context and on the other hand hardcoding the fact that `NOTHING` has happened...

I suspect that we will need to clarify the lifecycle of this API before anyone tries to really implement it because there will be sizeable API breaks in the future.